File: api/webhook.py
from flask import Flask, request, jsonify
from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup, WebAppInfo
from telegram.ext import Application, CommandHandler, ContextTypes

app = Flask(__name__)

TOKEN = '8686893037:AAF63iFUhicT13-aNnSbskvWp6_MrVBzJjc'
WEBAPP_URL = 'https://testik-gamma.vercel.app'

application = Application.builder().token(TOKEN).build()

def get_keyboard():
    keyboard = [
        [InlineKeyboardButton("📱 Открыть VRN", web_app=WebAppInfo(url=WEBAPP_URL))]
    ]
    return InlineKeyboardMarkup(keyboard)

async def start_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user = update.effective_user
    welcome_text = f"""Привет, {user.first_name}! 👋

VRN - лента новостей Воронежа.

Нажми кнопку ниже, чтобы открыть приложение."""
    await update.message.reply_text(welcome_text, reply_markup=get_keyboard())

async def help_command(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await update.message.reply_text("📌 /start - Главное меню")

application.add_handler(CommandHandler("start", start_command))
application.add_handler(CommandHandler("help", help_command))

@app.route('/api/webhook', methods=['POST'])
def webhook():
    import json
    update = Update.de_json(request.get_data(as_text=True), application.bot)
    application.run_async(application.process_update(update))
    return jsonify({'ok': True})

@app.route('/')
def index():
    return 'VRN Bot Webhook!'

@app.route('/api/health')
def health():
    return jsonify({'status': 'ok'})

File: api/webhook.js
module.exports = async (req, res) => {
  const TOKEN = '8686893037:AAF63iFUhicT13-aNnSbskvWp6_MrVBzJjc';
  const WEBAPP_URL = 'https://testik-gamma.vercel.app';

  if (req.method === 'GET') {
    return res.status(200).send('VRN Bot Webhook!');
  }

  if (req.method === 'POST') {
    try {
      const update = req.body;
      
      if (update.message) {
        const chat_id = update.message.chat.id;
        const text = update.message.text;
        
        if (text === '/start' || text === '/start@your_bot_username') {
          const keyboard = {
            inline_keyboard: [[{ text: '📱 Открыть VRN', web_app: { url: WEBAPP_URL } }]]
          };
          
          await fetch(`https://api.telegram.org/bot${TOKEN}/sendMessage`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
              chat_id,
              text: `Привет, ${update.message.from.first_name}! 👋\n\nVRN - лента новостей Воронежа.\n\nНажми кнопку ниже!`,
              reply_markup: keyboard
            })
          });
        } else if (text === '/help') {
          await fetch(`https://api.telegram.org/bot${TOKEN}/sendMessage`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
              chat_id,
              text: '📌 /start - Главное меню'
            })
          });
        }
      }
      
      return res.status(200).json({ ok: true });
    } catch (e) {
      return res.status(200).json({ ok: false, error: e.message });
    }
  }

  return res.status(200).send('VRN Bot!');
};
